src/index.js

```diff
@@ -0,0 +1,104 @@
+export default class Bank {
+    #accountID;
+    #transactionID;
+    constructor(name) {
+      this.name = name;
+      this.#accountID = 1;
+      this.#transactionID = 1;
+    }
+  
+    createAccount() {
+      return new Account(this.#accountID);
+      this.#accountID++;
+    }
+  
+    deposit(account, amount) {
+      let currentAccountBalance = account.balance;
+      let newAccountBalance = currentAccountBalance + amount;
+      let deposit = new Transaction(
+        this.#transactionID,
+        "10-01-2012",
+        amount,
+        0,
+        newAccountBalance
+      );
+      account.addTransaction(deposit);
+      this.#transactionID++;
+    }
+  
+    withdraw(account, amount) {
+      let currentAccountBalance = account.balance;
+      let newAccountBalance = currentAccountBalance - amount;
+      let withdraw = new Transaction(
+        this.#transactionID,
+        "10-01-2012",
+        0,
+        amount,
+        newAccountBalance
+      );
+      account.addTransaction(withdraw);
+      this.#transactionID++;
+    }
+  }
+  
+  export class Account {
+    #id;
+    #transactions = [];
+  
+    constructor(id) {
+      this.#id = id;
+    }
+  
+    addTransaction(transaction) {
+      if (!transaction instanceof Transaction) throw "unknown data";
+  
+      this.#transactions.unshift(transaction);
+    }
+  
+    printStatement(openingDate, closingDate) {
+      console.log("date       ||credit        ||debit     ||balance       ");
+      this.#transactions.forEach((transaction) => {
+        console.log(
+          `${transaction.date}        ||${transaction.credit}     ||${transaction.debit}      ||${transaction.balance}        `
+        );
+      });
+    }
+  
+    get balance() {
+      return this.#transactions.reduce(
+        (prev, curr, index) => prev + curr.credit - curr.debit,
+        0
+      );
+    }
+  }
+  
+  export class Transaction {
+    id;
+    date;
+    credit;
+    debit;
+    balance;
+  
+    constructor(id, date, credit, debit, balance) {
+      if (id === undefined || id === null || id === NaN) throw "id is required";
```

Comment:
`if(!id || isNaN(id))`

Reply:
thank you for these corrections, i will update this.